src/app/feedback/feedback.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from '../shared/user.service';
+import { BrowserModule } from '@angular/platform-browser';
+import { FormsModule }   from '@angular/forms';
+import { CouchService } from '../shared/couchdb.service';
+
+export class Feedback{
+  name:string
+  isUrgent:boolean;
+  feedbackType:string;
+  feedbackMsg:string;
+}
+
+@Component({
+  selector: 'feedback',
+  templateUrl: './feedback.component.html'
+})
+
+export class FeedbackComponent implements OnInit {
+  msgForUsr:string;
+  isFeedbackMsg :boolean=true;
+  feedback:Feedback= new Feedback();
+  message:string;
+  fedbkSubmitted:boolean= false;
+
+  constructor(
+    private userService: UserService,
+    private couchService:CouchService
+    ) { }
+
+  ngOnInit() {
+    this.feedback.name= this.userService.get().name;
+  }
+
+  submitfeedback(){
+    if (this.feedback.feedbackMsg===undefined|| this.feedback.feedbackMsg=="") {
```

Comment:
You could shorten it to if (!!urlFragment) { since it's a truthy value. More info: https://basarat.gitbooks.io/typescript/docs/tips/truthy.html Also please use the strict equality operator `===` instread of `==`.